### PR TITLE
Add a no-fail mode to the terminator

### PR DESCRIPTION
**Pull request by GiudGiud**
refs https://github.com/idaholab/moose/issues/28273

### CODE REVIEW

**Comment by GiudGiud** [comment]
@vincentlaboure

File: framework/src/userobjects/Terminator.C [hunk 1]
```diff
@@ -112,12 +116,17 @@ Terminator::handleMessage()
     message = "Terminator '" + name() + "' is causing ";
     if (_fail_mode == FailMode::HARD)
       message += "the execution to terminate.\n";
-    else
+    else if (_fail_mode == FailMode::SOFT)
       message += "a time step cutback by marking the current step as failed.\n";
+    else
+      message = "";
```
Comment (vincentlaboure) [inline framework/src/userobjects/Terminator.C:122]: Not a huge deal, but wouldn't the message sound incomplete in that NONE case? ("Terminator '" + name() + "' is causing ")

**Comment by vincentlaboure** [comment]
Thanks, @GiudGiud !

File: framework/src/userobjects/Terminator.C (hunk 1 above)
Comment (GiudGiud) [reply framework/src/userobjects/Terminator.C:122]: I ll rework this

File: framework/include/userobjects/Terminator.h [hunk 2]
```diff
@@ -51,7 +51,7 @@ class Terminator : public GeneralUserObject
   /// handle output of the optional message
   void handleMessage();
 
-  const enum class FailMode { HARD, SOFT } _fail_mode;
+  const enum class FailMode { HARD, SOFT, NONE } _fail_mode;
   const enum class ErrorLevel { INFO, WARNING, ERROR, NONE } _error_level;
```
Comment (lindsayad) [inline framework/include/userobjects/Terminator.h:55]: these two enums have too similar of a name IMO. I think we need some detailed doxygen and/or change these enum names to further delineate what the different roles of these are

**Review by lindsayad — APPROVED** [review]

**Comment by GiudGiud** [comment]
Test failures are a fluke in compiling and a distributed failure in NS for mixture physics which we ll have to look at separately.

Thanks for the review Alex and thanks for the suggested feature Vincent!

**Pull request merged**